### PR TITLE
Deprecate shapes in `bevy_render::mesh::shape`

### PR DESCRIPTION
**Pull request by Jondolf**
# Objective

#11431 and #11688 implemented meshing support for Bevy's new geometric primitives. The next step is to deprecate the shapes in `bevy_render::mesh::shape` and to later remove them completely for 0.14.

## Solution

Deprecate the shapes and reduce code duplication by utilizing the primitive meshing API for the old shapes where possible.

Note that some shapes have behavior that can't be exactly reproduced with the new primitives yet:

- `Box` is more of an AABB with min/max extents
- `Plane` supports a subdivision count
- `Quad` has a `flipped` property

These types have not been changed to utilize the new primitives yet.

---

## Changelog

- Deprecated all shapes in `bevy_render::mesh::shape`
- Changed all examples to use new primitives for meshing

## Migration Guide

Bevy has previously used rendering-specific types like `UVSphere` and `Quad` for primitive mesh shapes. These have now been deprecated to use the geometric primitives newly introduced in version 0.13.

Some examples:

```rust
let before = meshes.add(shape::Box::new(5.0, 0.15, 5.0));
let after = meshes.add(Cuboid::new(5.0, 0.15, 5.0));

let before = meshes.add(shape::Quad::default());
let after = meshes.add(Rectangle::default());

let before = meshes.add(shape::Plane::from_size(5.0));
// The surface normal can now also be specified when using `new`
let after = meshes.add(Plane3d::default().mesh().size(5.0, 5.0));

let before = meshes.add(
    Mesh::try_from(shape::Icosphere {
        radius: 0.5,
        subdivisions: 5,
    })
    .unwrap(),
);
let after = meshes.add(Sphere::new(0.5).mesh().ico(5).unwrap());
```

### CODE REVIEW

File: crates/bevy_render/src/mesh/shape/mod.rs [hunk 1]
```diff
@@ -27,6 +33,10 @@ impl From<Cube> for Mesh {
 }
 
 /// An axis-aligned box defined by its minimum and maximum point.
+#[deprecated(
+    since = "0.13.0",
+    note = "please use the `Cuboid` primitive for meshing or `Aabb2d` for a bounding volume"
+)]
 #[derive(Debug, Copy, Clone)]
 pub struct Box {
     pub min_x: f32,
```
Comment (NthTensor) [inline crates/bevy_render/src/mesh/shape/mod.rs:42]: Hooray!

**Review by alice-i-cecile — APPROVED** [review]
Nice deprecation notices :)

File: crates/bevy_render/src/mesh/primitives/dim2.rs [hunk 2]
```diff
@@ -405,3 +405,62 @@ impl From<Capsule2dMeshBuilder> for Mesh {
...
+                let round = (*coord * 2.).round() / 2.;
+                if (*coord - round).abs() < 0.00001 {
+                    *coord = round;
+                }
+            }
+        }
+    }
+
+    #[test]
+    fn test_regular_polygon() {
```
Comment (NthTensor) [inline crates/bevy_render/src/mesh/primitives/dim2.rs:429]: Document the purpose of this test please. I'm not sure I could fix this if I broke it.

File: errors/B0004.md [hunk 3]
```diff
@@ -33,7 +33,7 @@ fn setup_cube(
         .with_children(|parent| {
             // cube
             parent.spawn(PbrBundle {
-                mesh: meshes.add(shape::Cube { size: 1.0 }),
+                mesh: meshes.add(Cuboid::default()),
```
Comment (NthTensor) [inline errors/B0004.md:36]: Nice catch here.

**Review by NthTensor — APPROVED** [review]
Looks great. This must have been pretty tedious. Really appreceate how quickly you got this in.

File: crates/bevy_render/src/mesh/primitives/dim2.rs (hunk 2 above)
Comment (Jondolf) [reply crates/bevy_render/src/mesh/primitives/dim2.rs:429]: It's not my test, it was added by https://github.com/bevyengine/bevy/pull/11017. Personally, I find it to be a bit redundant and would remove it, but it was on the old `RegularPolygon` type so I kept it for now.

Comment (NthTensor) [reply crates/bevy_render/src/mesh/primitives/dim2.rs:429]: "Document RegularPolygon by writing a test" and then proceeds to write no documentation for the test.

Fair enough.

**Pull request merged**